### ISSUE TITLE
No cipher nor compression, ok for  renegotiation?

### ISSUE THREAD

**Issue by boaks** [eclipse/tinydtls#100]
in develop-branch, dtls.c, dtls_update_parameters: 

```
   /* Looks like we do not have a cipher nor compression. This is ok
     * for renegotiation, but not for the initial handshake. */
```

Logical/functional it seems to be OK for me.
But do you have also a RFC cite for that?

[RFC5246 - 7.4.1.2.  Client Hello](https://datatracker.ietf.org/doc/html/rfc5246#page-41)

```
CipherSuite cipher_suites<2..2^16-2>;
CompressionMethod compression_methods<1..2^8-1>;
```
FMPOV, that implies at least 2 and 1 byte must be used, empty would be `<0..`.
 
>  If the
>       session_id field is not empty (implying a session resumption
>       request), this vector MUST include at least the cipher_suite from
>       that session. 
> 
...
> If the session_id field is not empty
>       (implying a session resumption request), it MUST include the
>       compression_method from that session.  This vector MUST contain,
>       and all implementations MUST support, CompressionMethod.null.
>       Thus, a client and server will always be able to agree on a
>       compression method.

That refers to a resumption, so it may not apply. But I miss similar statements for renegotiation.

**Comment by boaks** [comment]
```
i = dtls_uint16_to_int(data);
if (data_length < i + sizeof(uint16)) {
```

The check above that comment, irritates me as well.
It doesn't check for `i==0`, what I would assume based on the comment, it checks, if the size of the cipher suites exceeds the available data! FMPOV, that must always fail.

**Comment by boaks** [comment]
My understanding of an "abbreviated/resumption handshake":

A client may try to resume, but though the server may fall back to a full-handshake, the client must at least offer all parameter to be able to fallback. That mostly means, the client sends the negotiated parameters.

PR #107 addresses that for the cipher suite and the compression method.

**Comment by boaks** [comment]
Fixed with PR #107
